sw/device/lib/dif/dif_aes.h

```diff
@@ -0,0 +1,359 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_AES_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_AES_H_
+
+#include "sw/device/lib/base/mmio.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * AES key length in bits.
+ */
+typedef enum dif_aes_key {
+  kDifAesKey128 = 0,
+  kDifAesKey192,
+  kDifAesKey256,
+} dif_aes_key_t;
+
+/**
+ * AES cipher modes.
+ */
+typedef enum dif_aes_mode {
+  kDifAesModeECB = 0, /**< The Electronic Codebook Mode. */
+  kDifAesModeCBC,     /**< The Cipher Block Chaining Mode. */
+  kDifAesModeCTR,     /**< The Counter Mode. */
+} dif_aes_mode_t;
+
+/**
+ * AES operation.
+ */
+typedef enum dif_aes_operation {
+  kDifAesOperationEncrypt = 0,
+  kDifAesOperationDecrypt,
+} dif_aes_operation_t;
+
+/**
+ * AES operation mode.
+ */
+typedef enum dif_aes_operation_mode {
+  kDifAesOperationModeAuto = 0,
+  kDifAesOperationModeManual,
+} dif_aes_operation_mode_t;
+
+/**
+ * AES configuration data.
+ */
+typedef struct dif_aes_config {
+  dif_aes_key_t key_len;
+  dif_aes_mode_t cipher_mode;
+  dif_aes_operation_t operation;
+  dif_aes_operation_mode_t operation_mode;
```

Comment:
I was itching to store these values in `dif_aes` after configuration (caching the current configuration for DIF internal use), to avoid a number of mmio operations. I see this more like in dif_spi where rx sizes are stored in `dif_aes`. However, for now I am checking the configuration values internally through mmio  operations. What do you think?